File: backend/src/main/java/com/slparcelauctions/backend/admin/infrastructure/bots/BotHeartbeatRequest.java
package com.slparcelauctions.backend.admin.infrastructure.bots;

import jakarta.validation.constraints.NotBlank;

import java.time.OffsetDateTime;

public record BotHeartbeatRequest(
        @NotBlank String workerName,
        @NotBlank String slUuid,
        @NotBlank String sessionState,
        String currentRegion,
        String currentTaskKey,
        String currentTaskType,
        OffsetDateTime lastClaimAt) {
}

File: bot/src/Slpa.Bot/Backend/Models/BotHeartbeatRequest.cs
namespace Slpa.Bot.Backend.Models;

/// <summary>
/// Mirrors the backend's BotHeartbeatRequest record. Serialized camelCase via
/// the shared JsonOpts. Backend requires workerName/slUuid/sessionState
/// non-blank; the rest are nullable.
/// </summary>
public sealed record BotHeartbeatRequest(
    string WorkerName,
    string SlUuid,
    string SessionState,
    string? CurrentRegion,
    string? CurrentTaskKey,
    string? CurrentTaskType,
    DateTimeOffset? LastClaimAt);
